src/catalog/lib.rs

```diff
@@ -58,6 +60,9 @@ pub struct Catalog {
     ambient_schemas: BTreeMap<String, Schema>,
     storage: Arc<Mutex<sql::Connection>>,
     serialize_item: fn(&CatalogItem) -> Vec<u8>,
+    // Used to uniquely identify any sink storage created during this catalog's
+    // lifetime
+    sink_suffix: Option<String>,
```

Comment:
I think the constituent parts are quite a bit more general than just the sink suffix, and denormalizing into the string seems unnecessarily limiting. Can we split this into its two constituent fields: `creation_time` and `nonce`?

Reply:
Also `creation_time` should be a `SystemTime`, so we can avoid the dependency on `chrono` in this package.